presto-main/src/main/java/com/facebook/presto/operator/SimplePagesHashStrategy.java

```diff
@@ -200,6 +200,16 @@ public boolean positionEqualsPosition(int leftBlockIndex, int leftPosition, int
     @Override
     public boolean applyFilterFunction(int leftBlockIndex, int leftPosition, int rightPosition, Block[] allRightBlocks)
     {
-        return filterFunction.get().filter(leftPosition, channelArrays.get(leftBlockIndex), rightPosition, allRightBlocks);
+        return filterFunction.get().filter(leftPosition, getLeftBlocks(leftBlockIndex), rightPosition, allRightBlocks);
+    }
+
+    private Block[] getLeftBlocks(int leftBlockIndex)
+    {
+        if (channelArrays.isEmpty()) {
+            return EMPTY_BLOCK_ARRAY;
```

Comment:
instead of putting this here, have the applyFilterFunction directly return the boolean if channelArrays.isEmpty()